src/App.tsx

```diff
@@ -1,61 +1,122 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const todosMapped: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [titleNew, setTitle] = useState<string>('');
+  const [userID, setUserID] = useState<string>('0');
+  const [todos, setTodos] = useState<Todo[]>(todosMapped);
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+
+  const maxId = Math.max(...todosMapped.map(todo => todo.id));
+  const addTodo = (formTitle: string, formUserId: string) => {
+    if (titleNew.trim().length !== 0 && +formUserId > 0) {
+      // todosMapped.push(newTodo);
+      setTitle('');
+      setUserID('0');
+      setTodos([
+        ...todos,
+        {
+          id: maxId + 1,
+          title: formTitle,
+          completed: false,
+          userId: +formUserId,
+          user: getUser(+formUserId),
+        },
```

Comment:
I think it would be better to create a variable for a new todo and pass it here to make the code a bit cleaner